Doc/library/tty.rst

```diff
@@ -1,41 +1,98 @@
-:mod:`tty` --- Terminal control functions
-=========================================
+:mod:`tty` --- Termios module extensions
+========================================
 
 .. module:: tty
    :platform: Unix
-   :synopsis: Utility functions that perform common terminal control operations.
+   :synopsis: Termios module extensions
 
 .. moduleauthor:: Steen Lumholt
 .. sectionauthor:: Moshe Zadka <moshez@zadka.site.co.il>
+.. sectionauthor:: Soumendra Ganguly <soumendra@tamu.edu>
 
 **Source code:** :source:`Lib/tty.py`
 
 --------------
 
-The :mod:`tty` module defines functions for putting the tty into cbreak and raw
-modes.
+The :mod:`tty` module extends the termios module by defining utility functions
+and classes.
 
 Because it requires the :mod:`termios` module, it will work only on Unix.
 
-The :mod:`tty` module defines the following functions:
+winsize Objects
+---------------
+
+.. data:: HAVE_WINSZ
+
+   :const:`True` if both :const:`termios.TIOCGWINSZ` and :const:`termios.TIOCSWINSZ`
+   are defined; :const:`False` otherwise. The :class:`winsize` objects are implemented
+   if and only if this is :const:`True`.
+
+
+.. class:: winsize(row=0, col=0, xpixel=0, ypixel=0, fd=None)
+
```

Comment:
@jonathanslenders requested a namedtuple. I have made this class which serves the same purpose + ensures that the attributes are nonnegative integers; the traditional struct winsize has `unsigned short` members; I have done that by defining a __setattr__; please check tty.py.